### PR TITLE
feature: notice

### PR DESCRIPTION
**Pull request by Tnixc**
Adds functionality for notices:

- pauses all on hover
- stroke is 2px for more emphasis, as the shadow is not very visible on most backgrounds. also matches with the timer bar
- I don't think we need to add a notification center at this stage. (literally the only use for notice rn is 'you shuffled the queue')

### CODE REVIEW

File: src/renderer/src/components/notice/Notice.tsx [hunk 1]
```diff
@@ -1,108 +1,155 @@
...
+    }
+  });
+
   return (
     <div
-      class={"notice-wrapper"}
-      onPointerEnter={pauseDrain}
-      onPointerLeave={() => setDrainTime(props.notice.timeoutMS ?? 10_000)}
+      ref={noticeRef}
+      class={twMerge(
```
Comment (duduBTW) [inline src/renderer/src/components/notice/Notice.tsx:96]: Can't we do:

```
const bgStyle = cva([
  "after:absolute after:inset-0 after:-z-20 after:size-20 after:rounded-full after:blur-2xl after:content-['']",
  "before:absolute before:inset-0.5 before:-z-10 before:rounded-[10px] before:bg-thick-material before:content-['']",
  "group relative w-96 transform overflow-hidden rounded-xl bg-thick-material p-4 shadow-2xl ring-2 ring-inset ring-stroke backdrop-blur-md duration-300 ease-in-out"
], {
  variants: {
    variant: {
      success: "bg-green after:bg-green",
      neutral: "bg-overlay after:bg-overlay",
      error: "bg-red after:bg-red",
    },
  },
  defaultVariants: {
    variant: "neutral",
  },
});

// ...

class={bgStyle({ variant: props.notice.variant, class: props.class })}
classList={{
  // Visibility styles
  "translate-y-0 opacity-100": isVisible(),
  "-translate-y-full opacity-0 blur-sm", !isVisible(),
  // Removing styles
  "my-0 h-0 max-h-0 min-h-0 -rotate-12 scale-75 py-0 opacity-0 blur-sm": isRemoving(),
  "my-2 min-h-20": !isRemoving()
}}
```

Comment (duduBTW) [reply src/renderer/src/components/notice/Notice.tsx:96]: Actually you do use the bgStyle somewhere else.

File: src/renderer/src/global.css [hunk 2]
```diff
@@ -70,6 +70,7 @@
 * {
   margin: 0;
   padding: 0;
+  user-select: none;
```
Comment (duduBTW) [inline src/renderer/src/global.css:73]: Should this change be here?

File: src/renderer/src/components/notice/Notice.tsx [hunk 3]
```diff
@@ -1,108 +1,155 @@
...
+          </div>
+        </Show>
+        <div class="overflow-hidden">
+          <Show when={props.notice.title}>
+            <h3 class="mb-1 overflow-hidden text-ellipsis whitespace-nowrap text-base font-semibold">
+              {props.notice.title}
+            </h3>
+          </Show>
+          <Show when={props.notice.description}>
+            <p class="overflow-hidden text-ellipsis whitespace-nowrap text-sm text-subtext">
```
Comment (duduBTW) [inline src/renderer/src/components/notice/Notice.tsx:135]: I don't think we should limit the number of lines on the notice description to only one line, 2 lines should be good. (`line-clamp-2`)

File: src/renderer/src/global.css (hunk 2 above)
Comment (Tnixc) [reply src/renderer/src/global.css:73]: I feel like it's a sane default for an electron app, and idk about making another PR just for this one line change. should have mentioned this though

Comment (brw) [reply src/renderer/src/global.css:73]: Like I mentioned before I'd rather not set this without either having a context menu option to copy certain text elements or by changing those elements (e.g. song title and artist) to be selectable.

**Review by duduBTW — APPROVED** [review]

**Pull request merged**